.github/workflows/lib-ci-dev.yaml

```diff
@@ -169,15 +169,18 @@ jobs:
       with:
         python-version: '3.10'
 
+    - name: Install uv
+      uses: astral-sh/setup-uv@v6
+
     - name: install dependencies
       run: |
-        python -m pip install --upgrade pip
-        pip install -q toml
-        pip install -q https://github.com/qiime2/q2lint/archive/master.zip
-        pip install -q flake8
+        uv venv
+        uv pip install -q toml https://github.com/qiime2/q2lint/archive/master.zip
 
-    - name: run flake8
-      run: flake8
+    - name: run ruff
+      uses: astral-sh/ruff-action@v3
+      with:
+        args: "check --output-format=github"
```

Comment:
ruff may need some configuration in order to honor the existing flake8 config:  https://docs.astral.sh/ruff/configuration/